src/db/commit.rs

```diff
@@ -185,15 +187,19 @@ impl Commit {
     pub async fn base_snapshot(
         hash: &str,
         dag_read: &dag::Read<'_>,
-    ) -> Result<Commit, FromHashError> {
-        use FromHashError::*;
-        let mut commit = Commit::from_hash(hash, dag_read).await?;
+    ) -> Result<Commit, BaseSnapshotError> {
+        use BaseSnapshotError::*;
+        let mut commit = Commit::from_hash(hash, dag_read)
+            .await
+            .map_err(NoSuchCommit)?;
         while !commit.meta().is_snapshot() {
             let meta = commit.meta();
             let basis_hash = meta
                 .basis_hash()
-                .ok_or_else(|| ChunkMissing("None".to_string()))?;
-            commit = Commit::from_hash(basis_hash, dag_read).await?;
+                .ok_or_else(|| NoBasis(format!("Commit {} has no basis", commit.chunk.hash())))?;
```

Comment:
Nit, but I do not think the formatted string is necessary here. You could just have the parameter be the hash of the chunk with no basis. The name of the type tells you what the problem is, and when the debug formatter gets its hand on it, it prints something like `NoBasis(<hash>)` by default. Not requesting a change here, just a stylistic suggestion for next time. All this error handling is evolving anyway.